### PR TITLE
test(solana): add Solana-to-Solana IBC integration tests

### PR DESCRIPTION
**Pull request by mariuszzak**
## Description

Add a dedicated `integration-tests` crate with Solana-to-Solana IBC integration tests that exercise the full packet lifecycle across independent `ProgramTest` chains. Unlike unit tests that verify individual instructions in isolation, these tests run the complete cross-chain flow end-to-end using real program CPI chains. See the [README](https://github.com/cosmos/solidity-ibc-eureka/blob/mariuszzak/solana-to-solana-integration-tests/programs/solana/integration-tests/README.md) for architecture diagrams, module overview and a test authoring guide.

### Test infrastructure

- Each test spins up one or more independent `ProgramTest` chains with a mock light client, focusing on correctness of the IBC state machine rather than proof verification
- **Actor-based design**:
  - `Deployer` — owns initialization and upgrade authority transfer
  - `Admin` — manages access manager operations
  - `IftAdmin` — manages IFT admin operations (pause, admin transfer, minting)
  - `User` — initiates sends
  - `Relayer` — uploads chunks and delivers packets
- Composable `ChainProgram` trait — each program struct (`TestIbcApp`, `Ics27Gmp`, `Ift`, etc.) owns its own registration, initialization and upgrade-authority logic
- Assertion helpers for commitment, receipt and error verification

### Coverage — 40 tests

- **Router** (15) — packet lifecycle, timeouts, replay protection, ordering edge cases, proof rejection
- **GMP** (12) — lifecycle, multi-user, timeout, CPI authorization, three-chain multi-hop
- **IFT** (7) — lifecycle, refunds, batch transfers, Token 2022, admin operations
- **Admin** (6) — access manager transfer for ICS26 Router and GMP

### Non-test changes

- Add `REJECT_PROOF` mechanism to `mock-light-client` for proof rejection tests
- Fix `test-ibc-app` `OnTimeoutPacket` CPI account layout — an unused `escrow_account` field caused account misalignment when called via the router

### CODE REVIEW

File: programs/solana/integration-tests/src/actors/mod.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+pub mod user;
+
+/// Shared interface for test actors (`Deployer`, `Admin`, `IftAdmin`, `User`, `Relayer`).
+pub trait Actor {
+    fn pubkey(&self) -> Pubkey;
+}
+
+/// Sign and submit a transaction with `keypair` as both fee payer and signer.
+async fn send_tx(
+    keypair: &Keypair,
```
Comment (srdtrk) [inline programs/solana/integration-tests/src/actors/mod.rs]: Maybe Actor can `send_tx` directly if we add `keypair()` to `Actor`

**Review by srdtrk — COMMENTED** [review]
right direction

File: programs/solana/integration-tests/src/actors/mod.rs (hunk 1 above)
Comment (mariuszzak) [reply programs/solana/integration-tests/src/actors/mod.rs]: good idea, applied!

File: programs/solana/integration-tests/tests/gmp/attestation_roundtrip.rs [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        programs,
+        lc_program_id: attestation::ID,
+    });
+    chain_a.prefund(&[&admin, &relayer, &user]);
+
+    let mut chain_b = Chain::new(ChainConfig {
+        client_id: "chain-b-client",
+        counterparty_client_id: "chain-a-client",
+        deployer: &deployer,
+        programs,
```
Comment (srdtrk) [inline programs/solana/integration-tests/tests/gmp/attestation_roundtrip.rs]: nit: would be good to test with separate attestor sets in each chains

File: programs/solana/integration-tests/tests/gmp/attestation_roundtrip.rs [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+    // ── Update client on both chains ──
+    relayer
+        .attestation_update_client(&mut chain_a, &attestors, PROOF_HEIGHT)
+        .await
+        .expect("update_client on A failed");
+    relayer
+        .attestation_update_client(&mut chain_b, &attestors, PROOF_HEIGHT)
+        .await
+        .expect("update_client on B failed");
```
Comment (srdtrk) [inline programs/solana/integration-tests/tests/gmp/attestation_roundtrip.rs]: Why do you always do this at the beginning of each test?

File: programs/solana/integration-tests/tests/gmp/three_chain.rs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+        &[ack_bc_ix],
+        Some(&relayer.pubkey()),
+        &[relayer.keypair()],
+        chain_b.blockhash(),
+    );
+    chain_b
+        .process_transaction(tx)
+        .await
+        .expect("B→C ack_packet failed");
+
```
Comment (srdtrk) [inline programs/solana/integration-tests/tests/gmp/three_chain.rs]: Add another leg for C -> A

**Review by srdtrk — COMMENTED** [review]
lgtm. Just left a few comment. Would be nice if all tests used attestor lc and not mock

File: programs/solana/integration-tests/tests/gmp/attestation_roundtrip.rs (hunk 3 above)
Comment (mariuszzak) [reply programs/solana/integration-tests/tests/gmp/attestation_roundtrip.rs]: The update_client is necessary because initialize only creates the client state with attestor config (latest_height = 0) but no consensus state accounts and recv_packet needs a consensus state at the proof height to verify membership proofs.

**Review by srdtrk — APPROVED** [review]
